# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1521227326 25200
#      Fri Mar 16 12:08:46 2018 -0700
# Branch stable
# Node ID 91331c76d9c1b395185055e86120f2c985aede95
# Parent  088506eadb44ecade8c0d81aa2f1f04e64af5801
invert the logic for whether to start the GUI in the octave wrapper

* main.in.cc (main): Set start_gui to false by default.  Handle the --gui
command-line option.

diff --git a/src/main.in.cc b/src/main.in.cc
--- a/src/main.in.cc
+++ b/src/main.in.cc
@@ -198,17 +198,17 @@ strsave (const char *s)
   return tmp;
 }
 
 int
 main (int argc, char **argv)
 {
   int retval = 0;
 
-  bool start_gui = true;
+  bool start_gui = false;
   bool gui_libs = true;
 
   set_octave_home ();
 
   std::string octave_bindir = get_octave_bindir ();
   std::string octave_archlibdir = get_octave_archlibdir ();
   std::string octave_cli
     = octave_bindir + dir_sep_char + "octave-cli-" OCTAVE_VERSION;
@@ -253,16 +253,25 @@ main (int argc, char **argv)
           // don't have to create a child process and wait for it to
           // exit.  But do exec "octave-gui", not "octave-cli", because
           // even if the --no-gui option is given, we may be asked to do
           // some plotting or ui* calls.
 
           start_gui = false;
           new_argv[k++] = argv[i];
         }
+      else if (! strcmp (argv[i], "--gui"))
+        {
+          // If we see this option, then we fork and exec octave with
+          // the --gui option, while continuing to handle signals in the
+          // terminal.
+
+          start_gui = true;
+          new_argv[k++] = argv[i];
+        }
       else if (! strcmp (argv[i], "--silent") || ! strcmp (argv[i], "--quiet"))
         {
           warn_display = false;
           new_argv[k++] = argv[i];
         }
       else if (! strcmp (argv[i], "--no-window-system"))
         {
           no_display = true;
